generators/server/templates/src/main/java/package/security/SecurityUtils.java.ejs

```diff
@@ -189,10 +203,33 @@ public final class SecurityUtils {
             claims.getOrDefault("roles", new ArrayList<>()));
     }
 
+    @SuppressWarnings("unchecked")
+    private static Collection<String> getScopesFromClaims(Map<String, Object> claims) {
+        for ( String attributeName : WELL_KNOWN_SCOPE_ATTRIBUTE_NAMES ) {
+            Object scopes = claims.get(attributeName);
+            if (scopes instanceof String) {
+                if (StringUtils.hasText((String) scopes)) {
+                    return Arrays.asList(((String) scopes).split(" "));
+                } else {
+                    return Collections.emptyList();
+                }
+            } else if (scopes instanceof Collection) {
+                return (Collection<String>) scopes;
+            }
+        }
+        return Collections.emptyList();
+    }
+
     private static List<GrantedAuthority> mapRolesToGrantedAuthorities(Collection<String> roles) {
         return roles.stream()
             .filter(role -> role.startsWith("ROLE_"))
             .map(SimpleGrantedAuthority::new).collect(Collectors.toList());
     }
+
+    private static List<GrantedAuthority> mapScopesToGrantedAuthorities(Collection<String> scopes) {
+        return scopes.stream()
+            .filter(scope -> scope.startsWith("SCOPE_"))
```

Comment:
are you sure scopes returned by IDP will have this prefix? Did you test this?

Reply:
Hello @vishal423, our implementation based on assumptions done in mapRolesToGrantedAuthorities() method. We've retested it and we found that scopes/roles returned from IDP don't have required prefix. Could you please confirm that this is a bug for mapRolesToGrantedAuthorities() method?

Reply:
@kozub, `mapRolesToGrantedAuthorities` is working as expected and aligned with roles/groups support on Keycloak/Okta. In that scenario, filtering is required to remove internal IDP specific roles/groups that are not used inside application.

For your use case, IDP wouldn't be returning scopes prefixed with `SCOPE_` and you need to add that manually to comply with Spring security `JwtAuthorityExtractor` implementation and this filtering wouldn't be required.